xdsl/irdl/irdl.py

```diff
@@ -458,7 +454,6 @@ def _irdl_list_to_attr_constraint(
 def irdl_to_attr_constraint(
     irdl: Any,
     *,
-    allow_type_var: bool = False,
```

Comment:
Why did you remove this? The reason I added this was because `isa(x, type_variable)` is a dangerous thing, and doesn't really mean anything if you don't have the "context" of the type variable.